### ISSUE TITLE
map_display_test fails intermittently

### ISSUE THREAD

**Issue by emersonknapp** [ros2/rviz#456]
Bug

This can be seen in the nightly armhf linux builds.
https://ci.ros2.org/view/nightly/job/nightly_linux-armhf_debug/202/testReport/junit/(root)/rviz_default_plugins/map_display_test_gtest_missing_result/

```
[ RUN      ] MapTestFixture.showMap_does_not_display_anything_if_transform_fails
map_display_test: /home/jenkins-agent/workspace/nightly_linux-armhf_debug/ws/build/rviz_ogre_vendor/ogre-v1.12.1-prefix/src/ogre-v1.12.1/OgreMain/src/OgreNode.cpp:377: void Ogre::Node::setPosition(const Vector3&): Assertion `!pos.isNaN() && "Invalid vector supplied as parameter"' failed.
```

This issue seems to already exist in rviz1 but is not being tracked in this repository https://github.com/ros-visualization/rviz/issues/811

**Comment by Martin-Idel** [comment]
Ogre doesn't initialize vectors and quaternions in the default constructor (this is a conscious choice for performance reasons).

I think here is the problem:
https://github.com/ros2/rviz/blob/c28f2c2798b6c3bc21486bf2e0572952e3eb875a/rviz_default_plugins/src/rviz_default_plugins/displays/map/map_display.cpp#L538-L552

We ask for a transform and this fills the vector/quaternion with the correct translation and rotation of the transformation. However, if it fails it probably (?) does nothing.
Regardless of whether it fails, we feed the vector to the `setPostion` function. So if the vector wasn't initialized by the constructor and wasn't changed by the transform function because it failed, it is still uninitialized when being fed into the `setPostion` function, which is undefined behaviour.

So my guess is that moving lines 551-552 inside the else block should fix the test. This was probably missed when porting and needs to be fixed in several displays...

**Comment by Martin-Idel** [comment]
@emersonknapp If I am correct about what's the issue, then this should have been fixed by #459

**Comment by emersonknapp** [comment]
Yes - https://ci.ros2.org/view/nightly/job/nightly_linux-armhf_debug/ appears to not be seeing this test failure any more. Thanks @Martin-Idel for following up!
